platform/platform-configuration-migration/src/test/java/org/codice/ddf/configuration/admin/ConfigurationAdminMigrationTest.java

```diff
@@ -257,106 +227,101 @@ public void testConstructorWithNullConfigurationFileExtension() {
                 null);
     }
 
-    @Test
-    public void testInitDoesNotCreateExistingProcessedAndFailedAndExportedDirectories()
-            throws IOException {
-        when(processedDirectory.exists()).thenReturn(true);
-        when(processedDirectory.isDirectory()).thenReturn(true);
-        when(failedDirectory.exists()).thenReturn(true);
-        when(failedDirectory.isDirectory()).thenReturn(true);
+    @Test(expected = IOException.class)
+    public void testInitDoesNotCreateExistingProcessedAndFailedDirectories() throws IOException {
 
-        ConfigurationAdminMigration configurationAdminMigration =
-                createConfigurationAdminMigratorWithNoFiles();
+        PowerMockito.doNothing()
+                .when(FileUtils.class);
+        FileUtils.forceMkdir(processedDirectoryPath.toFile());
+        PowerMockito.doThrow(new IOException())
+                .when(FileUtils.class);
+        FileUtils.forceMkdir(processedDirectoryPath.toFile());
+
+        PowerMockito.doNothing()
+                .when(FileUtils.class);
+        FileUtils.forceMkdir(failedDirectoryPath.toFile());
+        PowerMockito.doThrow(new IOException())
+                .when(FileUtils.class);
+        FileUtils.forceMkdir(failedDirectoryPath.toFile());
+
+        ConfigurationAdminMigration configurationAdminMigration = new ConfigurationAdminMigration(
+                configurationDirectoryStream,
+                processedDirectoryPath,
+                failedDirectoryPath,
+                configurationFileFactory,
+                configurationFilePoller,
+                configurationAdmin,
+                configurationFileExtension);
+
+        configurationAdminMigration.init();
         configurationAdminMigration.init();
 
-        verify(processedDirectory, never()).mkdir();
-        verify(failedDirectory, never()).mkdir();
         verify(configurationDirectoryStream).close();
```

Comment:
Since we need to verify that `configurationDirectoryStream` gets closed even when an exception gets thrown, you should use the `try/catch` pattern described here: http://stackoverflow.com/a/13224598, and call `verify()` inside the `catch` block.

If you want to keep the `@Test(expected=IOException.class)` annotation, you can skip the `fail()` in the `try` block and do this instead:

```
try {
    // Test code
} catch (IOException e) {
    // Verifies
    throw e;
}
```